### PR TITLE
fix(config): improve Yale Lock support

### LINKED ISSUE #1637: Add support for Lockwood (Assa Abloy) Keyless Wireless Deadbolt
**Which device is missing?**

Lockwood (Assa Abloy) Keyless Wireless Deadbolt (YRD240)

**What are the IDs?**

Manufacturer ID: unknown
Product type: unknown
Product ID: unknown

**Did you check if we really don't have it?**

- [x] It is not in the [configuration DB](https://devices.zwave-js.io/)
- [x] It was not [merged recently](https://github.com/zwave-js/node-zwave-js/pulls?q=is%3Apr+is%3Aclosed) or has a [pending PR](https://github.com/zwave-js/node-zwave-js/pulls?q=is%3Aopen+is%3Apr)
- [x] It is not in the [TODO list](https://github.com/zwave-js/node-zwave-js/issues/1600)

**Do you have a link to the PDF manual?**
https://www.lockweb.com.au/Local/AU/Technical%20Downloads/Lockwood/Secure%20Connect/Keyless%20Wireless%20Digital%20Deadbolt/Lockwood%20Keyless%20Wireless%20Digital%20Deadbolt%20User%20Guide.pdf

### PR DESCRIPTION
**Pull request by blhoward2**
Cleans up and improves Yale Lock support. Closes #1637 which we already supported anyway

### CODE REVIEW

**Comment by blhoward2** [comment]
@AlCalzone ready for review

File: packages/config/config/devices/0x0129/cp2-5-3_1.78.json [hunk 1]
```diff
@@ -18,8 +18,7 @@
 	"supportsZWavePlus": true,
 	"paramInformation": {
 		"1": {
-			"label": "Silent Mode on/off",
-			"description": "Changes speaker settigs",
+			"label": "Volune",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0129/cp2-5-3_1.78.json]:
```suggestion
			"label": "Volume",
```

File: packages/config/config/devices/0x0129/cp2-5-3_1.78.json [hunk 2]
```diff
@@ -53,18 +52,18 @@
 			"allowManualEntry": false,
 			"options": [
 				{
-					"label": "Auto Relock On",
+					"label": "Enable",
 					"value": -128
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0129/cp2-5-3_1.78.json]: We could make this `128` without the minus if we set `unsigned: true`, but I don't think this is necessary

File: packages/config/config/devices/0x0129/cp2-5-3_1.78.json [hunk 3]
```diff
@@ -85,97 +84,101 @@
...
+			"label": "Shutdown Time",
+			"description": "Number of times the device is locked out after the invalid retries are exceeded",
 			"valueSize": 1,
-			"minValue": 10,
-			"maxValue": 127,
-			"defaultValue": 10,
+			"minValue": 1,
+			"maxValue": 255,
+			"defaultValue": 30,
+			"unsigned": true,
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0129/cp2-5-3_1.78.json]: Are you sure? This seems to be the time the device stays locked.

File: packages/config/config/devices/0x0129/cp2-5-3_1.78.json [hunk 4]
```diff
@@ -85,97 +84,101 @@
...
-					"value": -128
+					"label": "Disable",
+					"value": 0
 				},
 				{
-					"label": "One Touch Locking Off",
-					"value": 0
+					"label": "Enable",
+					"value": 255
 				}
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0129/cp2-5-3_1.78.json:146]: Are you sure? -128 is not equivalent to 0 or 255 in signed representation.

File: packages/config/config/devices/0x0129/cp2-5-3_1.78.json [hunk 5]
```diff
@@ -85,97 +84,101 @@
...
-					"label": "LED On",
-					"value": -128
+					"label": "Disable",
+					"value": 0
 				},
 				{
-					"label": "LED Off",
-					"value": 0
+					"label": "Enable",
+					"value": 255
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0129/cp2-5-3_1.78.json:166]: dito

File: packages/config/config/devices/0x0129/cp2-5-3_1.78.json [hunk 6]
```diff
@@ -85,97 +84,101 @@
...
-					"value": -128
+					"label": "Normal Operation",
+					"value": 0
 				},
 				{
-					"label": "Reset to Factory Off",
-					"value": 0
+					"label": "Reset Device",
+					"value": 1
 				}
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0129/cp2-5-3_1.78.json]: dito

File: packages/config/config/devices/0x0129/sr-bx-zw.json [hunk 7]
```diff
@@ -16,9 +16,83 @@
 		"min": "0.0",
 		"max": "255.255"
 	},
+	"paramInformation": {
+		"1": {
+			"label": "Alarm Length",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0129/sr-bx-zw.json]:
```suggestion
			"label": "Alarm Duration",
```
?

File: packages/config/config/devices/0x0129/sr-bx-zw.json [hunk 8]
```diff
@@ -16,9 +16,83 @@
...
+					"value": 0
+				},
+				{
+					"label": "Enable",
+					"value": 1
+				}
+			]
+		},
+		"3": {
+			"label": "Comfort LED",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0129/sr-bx-zw.json]: That label is not very self-explanatory. Add description maybe?

**Review by AlCalzone — DISMISSED** [review]

**Comment by towerhand** [comment]
I'm sorry for requesting it here in the PR but since this is a Yale lock batch I figured it would make sense. Can you please add the config for the YRD226? Right know it's getting recognized as a YRD256 so maybe only need to change the "label": "YRD256" to "YRD226 / YRD256"?
https://products.z-wavealliance.org/products/1878?selectedFrequencyId=2

Thanks for all your hard work on the database @blhoward2

File: packages/config/config/devices/0x0129/cp2-5-3_1.78.json (hunk 2 above)
Comment (blhoward2) [reply packages/config/config/devices/0x0129/cp2-5-3_1.78.json]: So just add "unsigned": true and change the -128 to 128?

File: packages/config/config/devices/0x0129/cp2-5-3_1.78.json (hunk 4 above)
Comment (blhoward2) [reply packages/config/config/devices/0x0129/cp2-5-3_1.78.json:146]: While this one file purports it to be -128/0, the text says it is 0XFF = on. Thoughts?

Screen Shot 2021-02-08 at 9 55 53 PM

OpenHAB also has it as 0/255:

Screen Shot 2021-02-08 at 9 57 12 PM

And the manufacturer's manual for these locks only has hex:

Screen Shot 2021-02-08 at 9 58 19 PM

File: packages/config/config/devices/0x0129/cp2-5-3_1.78.json (hunk 6 above)
Comment (blhoward2) [reply packages/config/config/devices/0x0129/cp2-5-3_1.78.json]: This also appears to have been wrong in the files:

Screen Shot 2021-02-08 at 9 59 47 PM

**Comment by blhoward2** [comment]
> I'm sorry for requesting it here in the PR but since this is a Yale lock batch I figured it would make sense. Can you please add the config for the YRD226? Right know it's getting recognized as a YRD256 so maybe only need to change the "label": "YRD256" to "YRD226 / YRD256"?
> https://products.z-wavealliance.org/products/1878?selectedFrequencyId=2
> 
> Thanks for all your hard work on the database @blhoward2

Done

File: packages/config/config/devices/0x0129/cp2-5-3_1.78.json (hunk 4 above)
Comment (AlCalzone) [reply packages/config/config/devices/0x0129/cp2-5-3_1.78.json:146]: Lets trust the text then (0/255)

File: packages/config/config/devices/0x0129/cp2-5-3_1.78.json [hunk 9]
```diff
@@ -43,32 +43,34 @@
...
-					"label": "Auto Relock On",
-					"value": -128
+					"label": "Enable",
+					"value": 128
 				},
 				{
-					"label": "Auto Relock Off",
+					"label": "Disable",
 					"value": 0
 				}
```
Comment (AlCalzone) [inline packages/config/config/devices/0x0129/cp2-5-3_1.78.json]: What does OpenHAB (or the manual text) have here? -128/0 seems suspicous.

Comment (blhoward2) [reply packages/config/config/devices/0x0129/cp2-5-3_1.78.json]: I can't actually find any manual with this in it, as it is a module that goes inside other locks, but the others are consistently 0/255 so I'm just going to go with that.

**Review by AlCalzone — APPROVED** [review]

**Pull request merged**